pkg/pgmodel/view_test.go

```diff
@@ -167,8 +167,164 @@ func TestSQLView(t *testing.T) {
 					}
 				}
 			}
-			getViewRowCount(t, db, "prom_series.\""+name+"\"", seriesCount)
-			getViewRowCount(t, db, "prom_metric.\""+name+"\"", pointCount)
+			getViewRowCount(t, db, "prom_series.\""+name+"\"", "", seriesCount)
+			getViewRowCount(t, db, "prom_metric.\""+name+"\"", "", pointCount)
+		}
+	})
+}
+
+func TestSQLViewSelectors(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	withDB(t, *database, func(db *pgxpool.Pool, t *testing.T) {
+		metrics := []prompb.TimeSeries{
+			{
+				Labels: []prompb.Label{
+					{Name: metricNameLabelName, Value: "cpu_usage"},
+					{Name: "namespace", Value: "production"},
+					{Name: "node", Value: "brain"},
+				},
+				Samples: []prompb.Sample{
+					{Timestamp: 10, Value: 0.5},
+					{Timestamp: 40, Value: 0.6},
+				},
+			},
+			{
+				Labels: []prompb.Label{
+					{Name: metricNameLabelName, Value: "cpu_usage"},
+					{Name: "namespace", Value: "dev"},
+					{Name: "node", Value: "pinky"},
+				},
+				Samples: []prompb.Sample{
+					{Timestamp: 10, Value: 0.1},
+					{Timestamp: 45, Value: 0.2},
+				},
+			},
+			{
+				Labels: []prompb.Label{
+					{Name: metricNameLabelName, Value: "cpu_usage"},
+					{Name: "namespace", Value: "dev"},
+					{Name: "node", Value: "brain"},
+					{Name: "new_tag", Value: "value"},
+				},
+				Samples: []prompb.Sample{
+					{Timestamp: 60, Value: 0.2},
+				},
+			},
+			{
+				Labels: []prompb.Label{
+					{Name: metricNameLabelName, Value: "cpu_total"},
+					{Name: "namespace", Value: "production"},
+					{Name: "node", Value: "brain"},
+				},
+				Samples: []prompb.Sample{
+					{Timestamp: 10, Value: 0.5},
+					{Timestamp: 40, Value: 0.6},
+				},
+			},
+			{
+				Labels: []prompb.Label{
+					{Name: metricNameLabelName, Value: "cpu_total"},
+					{Name: "namespace", Value: "dev"},
+					{Name: "node", Value: "pinky"},
+				},
+				Samples: []prompb.Sample{
+					{Timestamp: 10, Value: 0.1},
+					{Timestamp: 45, Value: 0.2},
+				},
+			},
+		}
+
+		ingestor := NewPgxIngestor(db)
+		defer ingestor.Close()
+		_, err := ingestor.Ingest(metrics)
+
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		queries := []struct {
+			where string
+			rows  int
+		}{
+			{
+				where: "WHERE labels ? ('namespace' == 'dev')",
+				rows:  3,
+			},
+			{
+				where: "WHERE labels ? ('namespace' !== 'dev')",
+				rows:  2,
+			},
+			{
+				where: "WHERE labels ? ('namespace' ==~ 'de.*')",
```

Comment:
Should we do some tests with explain on and see if we can get things to inline and get initplans?

Reply:
I think a lateral join from the metric table with eq to the series table would also be a very useful explain test / I'm just curious how that one looks. The explain stuff can probably just be in json format and the comparison should be relatively easy then? Not sure how PGX handles that output though.